src/pyFAI/gui/pilx/utils.py

```diff
@@ -74,12 +76,22 @@ def get_dataset(parent: h5py.Group | h5py.File, path: str) -> h5py.Dataset:
     return dset
 
 
-def get_radial_dataset(parent: h5py.Group, nxdata_path: str) -> h5py.Dataset:
+def get_radial_dataset(parent: h5py.Group, nxdata_path: str, size: Optional[int]=None) -> h5py.Dataset:
     nxdata = parent[nxdata_path]
     assert isinstance(nxdata, h5py.Group)
     assert nxdata.attrs["NX_class"] == "NXdata"
     axes = nxdata.attrs["axes"]
-    radial_path = axes[-1] if isinstance(axes, Iterable) else axes
+    if isinstance(axes, Iterable):
+        radial_path = axes[0]
+        if size is not None:
+            for idx in [0, -1, 1, -2]:
```

Comment:
If we can check for the size, what about going through all `axes` elements instead of these specific indices ?

Reply:
I intent to make the default view (in the Nexus sense) on the `map` array, not on the `intensity` array. This is to try to ensure the different axes have the proper shape while keeping the compatibility with former files.

Reply:
Let's check how pilx reacts with files produced with former version of pyFAI ...